File: ai-services/constants/file_extensions.py
#!/usr/bin/env python3
"""
Zentrale Konstanten für Dateiendungen
Stellt einheitliche Listen für Audio- und Video-Extensions bereit
"""

# Audio-Extensions
AUDIO_EXTENSIONS = [
    '.mp3',
    '.wav', 
    '.flac',
    '.m4a',
    '.aac',
    '.ogg'
]

# Video-Extensions
VIDEO_EXTENSIONS = [
    '.mp4',
    '.avi',
    '.mkv',
    '.mov',
    '.wmv',
    '.webm',
    '.mpg',
    '.mpeg',
    '.flv'
]

# Alle unterstützten Media-Extensions
MEDIA_EXTENSIONS = AUDIO_EXTENSIONS + VIDEO_EXTENSIONS

# Lyrics-Extensions
LYRICS_EXTENSIONS = [
    '.txt',
    '.json'
]

# Cover-Extensions
COVER_EXTENSIONS = [
    '.jpg',
    '.jpeg',
    '.png',
    '.gif'
]

# Hilfsfunktionen
def is_audio_file(filename: str) -> bool:
    """Prüft ob eine Datei eine Audio-Datei ist"""
    return any(filename.lower().endswith(ext) for ext in AUDIO_EXTENSIONS)

def is_video_file(filename: str) -> bool:
    """Prüft ob eine Datei eine Video-Datei ist"""
    return any(filename.lower().endswith(ext) for ext in VIDEO_EXTENSIONS)

def is_media_file(filename: str) -> bool:
    """Prüft ob eine Datei eine Media-Datei ist"""
    return any(filename.lower().endswith(ext) for ext in MEDIA_EXTENSIONS)

def is_lyrics_file(filename: str) -> bool:
    """Prüft ob eine Datei eine Lyrics-Datei ist"""
    return any(filename.lower().endswith(ext) for ext in LYRICS_EXTENSIONS)

def is_cover_file(filename: str) -> bool:
    """Prüft ob eine Datei eine Cover-Datei ist"""
    return any(filename.lower().endswith(ext) for ext in COVER_EXTENSIONS)

File: utils/fileExtensions.js
/**
 * Zentrale Konstanten für Dateiendungen
 * Stellt einheitliche Listen für Audio- und Video-Extensions bereit
 */

// Audio-Extensions
const AUDIO_EXTENSIONS = [
    '.mp3',
    '.wav',
    '.flac',
    '.m4a',
    '.aac',
    '.ogg'
];

// Video-Extensions
const VIDEO_EXTENSIONS = [
    '.mp4',
    '.avi',
    '.mkv',
    '.mov',
    '.wmv',
    '.webm',
    '.mpg',
    '.mpeg',
    '.flv'
];

// Alle unterstützten Media-Extensions
const MEDIA_EXTENSIONS = [...AUDIO_EXTENSIONS, ...VIDEO_EXTENSIONS];

// Lyrics-Extensions
const LYRICS_EXTENSIONS = [
    '.txt',
    '.json'
];

// Cover-Extensions
const COVER_EXTENSIONS = [
    '.jpg',
    '.jpeg',
    '.png',
    '.gif'
];

// Hilfsfunktionen
function isAudioFile(filename) {
    return AUDIO_EXTENSIONS.some(ext => filename.toLowerCase().endsWith(ext));
}

function isVideoFile(filename) {
    return VIDEO_EXTENSIONS.some(ext => filename.toLowerCase().endsWith(ext));
}

function isMediaFile(filename) {
    return MEDIA_EXTENSIONS.some(ext => filename.toLowerCase().endsWith(ext));
}

function isLyricsFile(filename) {
    return LYRICS_EXTENSIONS.some(ext => filename.toLowerCase().endsWith(ext));
}

function isCoverFile(filename) {
    return COVER_EXTENSIONS.some(ext => filename.toLowerCase().endsWith(ext));
}

module.exports = {
    AUDIO_EXTENSIONS,
    VIDEO_EXTENSIONS,
    MEDIA_EXTENSIONS,
    LYRICS_EXTENSIONS,
    COVER_EXTENSIONS,
    isAudioFile,
    isVideoFile,
    isMediaFile,
    isLyricsFile,
    isCoverFile
};
